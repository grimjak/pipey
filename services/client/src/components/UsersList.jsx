import React from 'react';

const UsersList = (props) => {
    return (
        <div>
            {
                props.users.map((user) => {
                    return (
                        <h4
                            key={user.id}
                            className="box title is-4"
                        >{ user.firstname } { user.lastname }
                        </h4>
                    )
                })
            }
        </div>
    )
};

export default UsersList;