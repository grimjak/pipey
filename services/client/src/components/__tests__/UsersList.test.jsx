import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import UsersList from '../UsersList';

const users = [
    {
        "username": "tb",
        "firstname": "ted",
        "lastname": "bear",
        "employeenumber": "1",
        "address": "23 blodsfsdf"
    },
    {
        "username": "bh",
        "firstname": "bob",
        "lastname": "holmes",
        "employeenumber": "2",
        "address": "77 verulam road"
    }
];

test('UsersList renders properly', () => {
    const wrapper = shallow(<UsersList users={users}/>);
    const element = wrapper.find('h4');
    expect(element.length).toBe(2);
    expect(element.get(0).props.children[0]).toBe('tb');
});

test('UserList renders a snapshot properly', () => {
    const tree = renderer.create(<UsersList users={users}/>).toJSON();
    expect(tree).toMatchSnapshot();
});