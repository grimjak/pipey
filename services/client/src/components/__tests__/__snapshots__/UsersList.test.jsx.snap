// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders a snapshot properly 1`] = `
<div>
  <h1
    className="title is-1"
  >
    All Users
  </h1>
  <hr />
  <br />
  <table
    className="table is-hoverable is-fullwidth"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Username
        </th>
        <th>
          Active
        </th>
        <th>
          Admin
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td />
        <td>
          tb
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
      </tr>
      <tr>
        <td />
        <td>
          bh
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
